src/common/query/src/logical_plan.rs

```diff
@@ -71,6 +74,39 @@ pub fn create_aggregate_function(
     )
 }
 
+/// Rename columns by applying a new projection. Returns an error if the column to be
+/// renamed does not exist. The `renames` parameter is a `Vector` with elements
+/// in the form of `(old_name, new_name)`.
+pub fn rename_logical_plan_columns(
+    enable_ident_normalization: bool,
+    plan: LogicalPlan,
+    renames: Vec<(&str, &str)>,
+) -> DatafusionResult<LogicalPlan> {
+    let mut projection = Vec::with_capacity(renames.len());
+
+    for (old_name, new_name) in renames {
+        let old_column: Column = if enable_ident_normalization {
+            Column::from_qualified_name(old_name)
+        } else {
+            Column::from_qualified_name_ignore_case(old_name)
```

Comment:
Is it guaranteed that the old name is always qualified?

Reply:
looks like all of them are unqualified names

<img width="440" alt="image" src="https://github.com/GreptimeTeam/greptimedb/assets/15380403/16e01f8f-0604-4c4a-9474-eee57b2ef7ea">